test/test_linalg.py

```diff
@@ -2295,6 +2295,18 @@ def test_eig_errors_and_warnings(self, device, dtype):
             with self.assertRaisesRegex(RuntimeError, "tensors to be on the same device"):
                 torch.linalg.eig(a, out=(out_w, out_v))
 
+    @skipCPUIfNoLapack
+    @skipCUDAIfNoMagma
+    @dtypes(*floating_and_complex_types())
+    def test_eig_with_nan(self, device, dtype):
+        for val in [np.inf, np.nan]:
```

Comment:
Is it allright to use numpy here? Afaik numpy is no longer in the dependency, right? Probably it is better to replace it with `float('inf')` and `float('nan')`. Also, I am not sure this approach works well with float32, as default floating type for numpy/python is double.

Reply:
That being said, I think the `float` solution is better, as `np` is not necessary here. Just a thought tho, not even a nit.

Reply:
Yet I am not sure this test is correct for 32-bit floats.

Reply:
NumPy is no longer a dependency of PyTorch itself but we do require it to run the tests, so yes you're OK to use NumPy unconditionally in the test suite